### PR TITLE
Add mentoring notes for scala sum-of-multiples

### PR DESCRIPTION
**Pull request by ricemery**
Add mentoring notes for scala sum-of-multiples.

First attempt add adding mentor notes.. Please let me know if anything is messed up.

Thx

### CODE REVIEW

File: tracks/sum-of-multiples/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    factors flatMap(x => x until limit by x) sum
+}
+```
+
+
+### Common suggestions
+
+- As an additional challenge, this exercise can be solved without using a local mutable var. If instead of using the for comprehension, filter over the (1 until limit) range, then checking if there exists n that are divisible by one of the factors.
+- As an additional challenge, this exercise can be solved without using recursion. If instead of using recursion, 'filter' over the (1 until limit) range, then checking if there 'exists' n that are divisible by one of the factors. 
+- Did you know that this solution does not pass all the unit test cases? Note that most of the test cases are marked pending. And, it is up to you to turn the test cases on as you develop your solution.
```
Comment (ErikSchierboom) [inline tracks/sum-of-multiples/mentoring.md]: I'm not sure about this section. Is this something to happens a lot when mentoring, that students have to be told that they have to modify the unit test cases?

File: tracks/sum-of-multiples/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+### Reasonable solutions
+
+```scala
```
Comment (ErikSchierboom) [inline tracks/sum-of-multiples/mentoring.md]: Just a suggestion. In the C# track, I use a sub-header to describe the various reasonable solutions if there is more than one (what makes them unique). For an example, see [the notes](https://github.com/exercism/website-copy/blob/master/tracks/csharp/exercises/bob/mentoring.md). You could consider adding those, but feel free to ignore.

File: tracks/sum-of-multiples/mentoring.md (hunk 1 above)
Comment (ricemery) [reply tracks/sum-of-multiples/mentoring.md]: Yes. If I had to estimate, the full suite of unit tests have not been run about 10% of the time.

File: tracks/sum-of-multiples/mentoring.md (hunk 2 above)
Comment (ricemery) [reply tracks/sum-of-multiples/mentoring.md]: Thanks!

File: tracks/sum-of-multiples/mentoring.md (hunk 1 above)
Comment (ErikSchierboom) [reply tracks/sum-of-multiples/mentoring.md]: Oh wow, that's huge!

**Pull request merged**